mantidproject/mantid
# Apply flood workspace algorithm

To be merged after https://github.com/mantidproject/mantid/pull/23317 and https://github.com/mantidproject/mantid/pull/23337

**Description of work.**
Added algorithm ApplyFloodWorkspace.

**To test:**
Run this script:
```
flood = CreateFloodWorkspace('OFFSPEC00035946.nxs', StartSpectrumIndex=250, EndSpectrumIndex=600,
                             Exclude=[260, 261, 262, 516, 517, 518], OutputWorkspace='flood')
data = Load('OFFSPEC00035946.nxs')
ApplyFloodWorkspace(InputWorkspace='data', FloodWorkspace='flood', OutputWorkspace='out')
```
If plotted the first bin of "out" workspace should have a straight line in the middle of the graph.

Fixes #23318 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
